File: backend/IMS_API/models/PurchaseOrder.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
namespace IMS_API.Models;

public class PurchaseOrder
{
    public int Id { get; set; }
    [Required, MaxLength(30)]
    public string OrderNumber { get; set; } = null!;
    public int SupplierId { get; set; }
    public Supplier Supplier { get; set; } = null!;
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public DateTime? ExpectedDeliveryDate { get; set; }
    public bool IsReceived { get; set; } = false;

    [Precision(18, 2)]
    public decimal TotalAmount { get; set; } = 0;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    public ICollection<PurchaseOrderItem> Items { get; set; } = new List<PurchaseOrderItem>();
}

File: frontend/IMS_WEB/src/app/models/puchaseorders.ts/purchaseorder.ts
export interface PurchaseOrder {
  id: number;
  orderNumber: string;

  supplierId: number;
  supplierName?: string;

  orderDate: Date;
  expectedDeliveryDate?: Date | null;

  isReceived: boolean;
  totalAmount: number;

  createdAt?: Date;
  updatedAt?: Date | null;

  items?: PurchaseOrderItem[];
}

export interface PurchaseOrderCreate {
  purchaseOrder: {
    orderNumber: string;
    supplierId: number;
    orderDate: Date;
    expectedDeliveryDate?: Date | null;
  };
  items: {
    productId: number;
    quantity: number;
    unitPrice: number;
    warehouseId: number;
  }[];
}

export interface PurchaseOrderWithItems {
  purchaseOrder: PurchaseOrder;
  inventoryItems: PurchaseOrderItem[];
}

export interface PurchaseOrderItem {
  id: number;
  purchaseOrderId: number;

  productId: number;
  productName?: string;

  quantity: number;
  unitPrice: number;

  warehouseId: number;
  warehouseName?: string;

  status: string;

  createdAt: Date;
  updatedAt?: Date | null;
}
